converter/types.go

```diff
@@ -29,36 +44,117 @@ func CreateBundleConfig(b *bundle.Bundle) *BundleConfig {
 }
 
 // PrepareForPush serializes a bundle config, generates its image manifest, and its manifest descriptor
-func (c *BundleConfig) PrepareForPush() (blob []byte, manifest []byte, blobDescriptor ocischemav1.Descriptor, manifestDescriptor ocischemav1.Descriptor, err error) {
-	bytes, err := json.Marshal(c)
+func (c *BundleConfig) PrepareForPush() (*PreparedBundleConfig, error) {
+	blob, err := json.Marshal(c)
+	if err != nil {
+		return nil, err
+	}
+	manifest := ocischemav1.Manifest{
+		Versioned: ocischema.Versioned{
+			SchemaVersion: 2,
+		},
+		Config: ocischemav1.Descriptor{
+			Digest:    digest.FromBytes(blob),
+			MediaType: CNABConfigMediaType,
+			Size:      int64(len(blob)),
+		},
+	}
 	if err != nil {
-		return nil, nil, ocischemav1.Descriptor{}, ocischemav1.Descriptor{}, err
+		return nil, err
 	}
+	manifestBytes, err := json.Marshal(&manifest)
+	if err != nil {
+		return nil, err
+	}
+
+	fallback, err := c.prepareForPushFallbackNoCustomMediaType(blob)
+	if err != nil {
+		return nil, err
+	}
+
+	return &PreparedBundleConfig{
+		ConfigBlob: blob,
+		ConfigBlobDescriptor: ocischemav1.Descriptor{
+			MediaType: CNABConfigMediaType,
+			Size:      int64(len(blob)),
+			Digest:    digest.FromBytes(blob),
+		},
+		Manifest: manifestBytes,
+		ManifestDescriptor: ocischemav1.Descriptor{
+			MediaType: ocischemav1.MediaTypeImageManifest,
+			Digest:    digest.FromBytes(manifestBytes),
+			Size:      int64(len(manifestBytes)),
+		},
+		Fallback: fallback,
+	}, nil
+}
+
+func (c *BundleConfig) prepareForPushFallbackNoCustomMediaType(blob []byte) (*PreparedBundleConfig, error) {
```

Comment:
I think this can be factorized with `PrepareForPush`, it's almost the same code, just need to configure the `MediaType` and a fallback functor.